presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,513 @@
+            return serialize(exteriorRing);
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Description("Returns a geometry that represents the shared portion of geometry A and geometry B")
+    @ScalarFunction("ST_Intersection")
+    @SqlType(NAME)
+    public static Slice stIntersection(@SqlType(NAME) Slice left, @SqlType(NAME) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        return serialize(leftGeometry.intersection(rightGeometry));
+    }
+
+    @Description("Returns a geometry that represents the portions of A and B that do not intersect")
+    @ScalarFunction("ST_SymDifference")
+    @SqlType(NAME)
+    public static Slice stSymmetricDifference(@SqlType(NAME) Slice left, @SqlType(NAME) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        return serialize(leftGeometry.symDifference(rightGeometry));
+    }
+
+    @SqlNullable
+    @Description("Returns TRUE if and only if no points of B lie in the exterior of A, and at least one point of the interior of B lies in the interior of A")
+    @ScalarFunction("ST_Contains")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stContains(@SqlType(NAME) Slice left, @SqlType(NAME) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        OperatorSimpleRelation containsOperator = OperatorContains.local();
+        return containsOperator.execute(leftGeometry.getEsriGeometry(), rightGeometry.getEsriGeometry(), leftGeometry.getEsriSpatialReference(), null);
+    }
+
+    @SqlNullable
+    @Description("Returns TRUE if the supplied geometries have some, but not all, interior points in common")
+    @ScalarFunction("ST_Crosses")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stCrosses(@SqlType(NAME) Slice left, @SqlType(NAME) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        OperatorSimpleRelation crossesOperator = OperatorCrosses.local();
+        return crossesOperator.execute(leftGeometry.getEsriGeometry(), rightGeometry.getEsriGeometry(), leftGeometry.getEsriSpatialReference(), null);
+    }
+
+    @SqlNullable
+    @Description("Returns TRUE if the Geometries do not spatially intersect - if they do not share any space together")
+    @ScalarFunction("ST_Disjoint")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stDisjoint(@SqlType(NAME) Slice left, @SqlType(NAME) Slice right)
+    {
+        OGCGeometry leftGeometry = deserialize(left);
+        OGCGeometry rightGeometry = deserialize(right);
+        OperatorSimpleRelation disjointOperator = OperatorDisjoint.local();
```

Comment:
why not leftGeometry.disjoint(rightGeometry) ? ditto ST_Intersects, ST_Overlaps, ST_Within, etc.